### PR TITLE
Triplet loss and norms benchmark on Cifar10

### PR DESCRIPTION
**Pull request by ClementWalter**

### CODE REVIEW

File: notebooks/triplet_loss_cifar10.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+train_dataset, val_dataset, test_dataset = tfds.load(
+    name="cifar10", split=["train[:90%]", "train[90%:]", "test"], as_supervised=True
+)
+train_dataset = train_dataset.shuffle(1024).batch(64, drop_remainder=True)
+val_dataset = val_dataset.shuffle(1024).batch(64, drop_remainder=True)
+test_dataset = test_dataset.batch(64, drop_remainder=True)
+
+train_labels = [batch[1].numpy().tolist() for batch in train_dataset]
+val_labels = [batch[1].numpy().tolist() for batch in val_dataset]
+test_labels = [batch[1].numpy().tolist() for batch in test_dataset]
```
Comment (Wirg) [inline notebooks/triplet_loss_cifar10.py:34]: une remarque tout à fait optionnelle. Je trouve `[labels.numpy().tolist() for _, labels in train_dataset]` plus clair.

File: notebooks/triplet_loss_cifar10.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+from keras_fsl.layers import GramMatrix
+from keras_fsl.losses.gram_matrix_losses import binary_crossentropy, class_consistency_loss, triplet_loss
+from keras_fsl.metrics.gram_matrix_metrics import classification_accuracy
+from keras_fsl.utils.tensors import get_dummies
+
+
+#%% Build datasets
+train_dataset, val_dataset, test_dataset = tfds.load(
+    name="cifar10", split=["train[:90%]", "train[90%:]", "test"], as_supervised=True
```
Comment (Wirg) [inline notebooks/triplet_loss_cifar10.py:26]: c'est cool et en même temps un peu étrange de faire ça dans une string.

File: notebooks/triplet_loss_cifar10.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+)
+
+output_dir = Path("logs") / "triplet_loss_cifar10"
+output_dir.mkdir(exist_ok=True, parents=True)
+results = []
+
+#%% Save test labels for later visualization in projector https://projector.tensorflow.org/
+out_m = io.open(output_dir / "meta.tsv", "w", encoding="utf-8")
+for img, labels in tfds.as_numpy(test_dataset):
+    [out_m.write(str(x) + "\n") for x in labels]
```
Comment (Wirg) [inline notebooks/triplet_loss_cifar10.py:58]: on en a déjà parlé non ? une boucle for nesté serait mieux. Ou alors il y a peut être un write_liines ou un truc du genre.
Ou encore : `'\n'.join(map(str, labels))`.
Une autre option pourrait être d'utiliser pandas.

File: notebooks/triplet_loss_cifar10.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    .groupby("split")
+    .apply(lambda group: pd.get_dummies(group.label).agg("sum"))
+)
+
+output_dir = Path("logs") / "triplet_loss_cifar10"
+output_dir.mkdir(exist_ok=True, parents=True)
+results = []
+
+#%% Save test labels for later visualization in projector https://projector.tensorflow.org/
+out_m = io.open(output_dir / "meta.tsv", "w", encoding="utf-8")
```
Comment (Wirg) [inline notebooks/triplet_loss_cifar10.py:56]: tu dois pouvoir utiliser un context manage (with)

File: notebooks/triplet_loss_cifar10.py (hunk 3 above)
Comment (ClementWalter) [reply notebooks/triplet_loss_cifar10.py:58]: c'est copié/collé du notebook tensorflow je ne l'avais même pas lu en détail

File: notebooks/triplet_loss_cifar10.py (hunk 4 above)
Comment (ClementWalter) [reply notebooks/triplet_loss_cifar10.py:56]: pareil, ça vient du notebook, pas lu

File: notebooks/triplet_loss_cifar10.py (hunk 2 above)
Comment (ClementWalter) [reply notebooks/triplet_loss_cifar10.py:26]: on peut créer des objets tfds.Split mais pas sûr qu'on y gagne grand chose
